morl_baselines/multi_policy/pareto_q_learning/pql.py

```diff
@@ -216,19 +221,19 @@ def train(
                 self.avg_reward[state, action] += (reward - self.avg_reward[state, action]) / self.counts[state, action]
                 state = next_state
 
-            self.epsilon = max(self.final_epsilon, self.epsilon * self.epsilon_decay)
+                if self.log and self.global_step % log_every == 0:
+                    self.writer.add_scalar("global_step", self.global_step, self.global_step)
+                    pf = self._eval_all_policies()
+                    log_all_multi_policy_metrics(
+                        current_front=pf,
+                        hv_ref_point=eval_ref_point,
+                        reward_dim=self.reward_dim,
+                        global_step=self.global_step,
+                        writer=self.writer,
+                        ref_front=known_pareto_front,
+                    )
 
-            if self.log and episode % log_every == 0:
-                self.writer.add_scalar("global_step", self.global_step, self.global_step)
-                pf = self._eval_all_policies()
-                log_all_multi_policy_metrics(
-                    current_front=pf,
-                    hv_ref_point=eval_ref_point,
-                    reward_dim=self.reward_dim,
-                    global_step=self.global_step,
-                    writer=self.writer,
-                    ref_front=known_pareto_front,
-                )
+            self.epsilon = max(self.final_epsilon, self.epsilon * self.epsilon_decay)
```

Comment:
I realized that we always use linearly decayed epsilon in the other algorithms, and here it uses exponentially decayed. Should we replace this by the linearly decayed so the exploration is the same?

Reply:
It is exponentially decaying in the original paper though. I think we should keep it like this.